src/io/mouse.js

```diff
@@ -1,20 +1,47 @@
 var MathUtil = require('../util/math-util');
 
-function Mouse () {
+function Mouse (runtime) {
     this._x = 0;
     this._y = 0;
     this._isDown = false;
+    /**
+     * Reference to the owning Runtime.
+     * Can be used, for example, to activate hats.
+     * @type{!Runtime}
+     */
+    this.runtime = runtime;
 }
 
 Mouse.prototype.postData = function(data) {
     if (data.x) {
-        this._x = data.x;
+        this._x = data.x - data.canvasWidth / 2;
     }
     if (data.y) {
-        this._y = data.y;
+        this._y = data.y - data.canvasHeight / 2;
     }
     if (typeof data.isDown !== 'undefined') {
         this._isDown = data.isDown;
+        if (this._isDown) {
+            this._activateClickHats(data.x, data.y);
+        }
+    }
+};
+
+Mouse.prototype._activateClickHats = function (x, y) {
```

Comment:
Hmm... that seems like a good idea at first glance. Can you think of any downsides?

Of course, thinking about that makes me wonder:
There's not a whole lot of difference between the object being sent using `postMessage` and the object being emitted by an `EventEmitter` (or similar). Would it be oversimplifying things to replace the VM side of the I/O device paradigm with something that translates certain posted messages into events emitted by a central "I/O Devices" event emitter?

I think that would work well for mouse and keyboard, but not for the clock. Are there other I/O devices that wouldn't work well with this model? If it works for everything except the clock then maybe the clock needs to be a different type of thing. On the other hand, I don't want to remove meaningful power or flexibility just because it makes the system simpler...